webproject-frontend/src/components/Themes.vue

```diff
@@ -0,0 +1,233 @@
+<!--
+
+    Webproject NG
+    Copyright © 2021 Gmasil
+
+    This file is part of Webproject NG.
+
+    Webproject NG is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Webproject NG is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Webproject NG. If not, see <https://www.gnu.org/licenses/>.
+
+-->
+<template>
+  <div>
+    <v-select
+      class="bg-theme-background-highlight text-theme-text"
+      v-model="selectedTheme"
+      :options="themes"
+      label="name"
+      @input="onThemeSelectionChange"
+    ></v-select>
+    <!--
+    <select v-model="selectedTheme" @change="onThemeSelectionChange">
+      <option v-for="theme in themes" v-bind:value="theme" :key="theme.id">
+        {{ theme.name }}
+      </option>
+    </select>
+    -->
+
+    <button @click="onActivateClick">Activate</button>
+    <button @click="onDuplicateClick">Duplicate</button>
+
+    <div v-if="selectedTheme && !selectedTheme.preset">
+      <div class="grid grid-cols-fit gap-x-4 gap-y-1 justify-items-start">
+        <span class="inline-block">Name</span>
+        <input type="text" v-model="selectedThemeCopy.name" />
+        <span class="inline-block">Background color</span>
+        <color-picker
+          class="border border-theme-text"
+          v-model="selectedThemeCopy.backgroundColor"
+          @change="onBackgroundColorChange"
+          disabled="true"
+        ></color-picker>
+        <span>Text color</span>
+        <color-picker
+          class="border border-theme-text"
+          v-model="selectedThemeCopy.textColor"
+        ></color-picker>
+      </div>
+
+      <button @click="onSaveActivateClick">Save and Activate</button>
+      <button @click="onSaveClick">Save</button>
+      <button @click="onResetClick">Reset</button>
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+  name: "Themes",
+  computed: {
+    currentUser() {
+      return this.$store.state.currentUser;
+    }
+  },
+  data() {
+    return {
+      themes: [],
+      selectedTheme: null,
+      selectedThemeCopy: null
+    };
+  },
+  created() {
+    axios
+      .get("/api/themes/available")
+      .then(response => {
+        this.themes = response.data;
+        if (this.themes.length > 0) {
+          this.themes.forEach(theme => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/gmasil/Webproject-NG/pullRequest?prid=8166316)
```suggestion
          this.themes.forEach((theme) => {
```